lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/codeactions/AddFeature.java

```diff
@@ -20,75 +20,62 @@
     public static final String FEATUREMANAGER_FORMAT = 
-            "<featureManager>" + NEW_LINE + FEATURE_FORMAT + NEW_LINE + "</featureManager>";
+            "\n\t<featureManager>" + 
+            "\n\t\t<feature>%s</feature>"  +
+            "\n\t</featureManager>";
 
     @Override
     public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeActions, CancelChecker cancelChecker) {
         Diagnostic diagnostic = request.getDiagnostic();
         DOMDocument document = request.getDocument();   
         try {
-            String insertText = NEW_LINE + FEATUREMANAGER_FORMAT;
-            DOMNode insertNode = document.getDocumentElement().getFirstChild();
-            Position insertPosition = XMLPositionUtility.createRange(insertNode.getStart(), insertNode.getEnd(), document).getStart();
+            String insertText = "";
+            DOMNode insertNode = null;
+            String indent = request.getXMLGenerator().getWhitespacesIndent();
+
             for (DOMNode node : document.getDocumentElement().getChildren()) {
                 if (LibertyConstants.FEATURE_MANAGER_ELEMENT.equals(node.getNodeName())) {
-                    insertText = NEW_LINE;
+                    // If featureManager is empty, add new feature after featureManager start tag.
+                    // Otherwise, add feature after last feature child.
                     insertNode = node.getLastChild();
-                    insertPosition = XMLPositionUtility.createRange(insertNode.getStart(), insertNode.getEnd(), document).getEnd();
-
-                    // document.getDocumentElement().getEndTagCloseOffset();
-                    // DOMElement de = (DOMElement) node;
-                    // Position start = document.positionAt(de.getStartTagCloseOffset()+1);
-                    // Position stop = document.positionAt(de.getEndTagCloseOffset());
-                    // Range targetRange = new Range(start, stop);
-                    // XMLGenerator generator = request.getXMLGenerator();                    
-                    String indent = request.getXMLGenerator().getWhitespacesIndent();
-                    int level = document.positionAt(((DOMElement)node).getStartTagOpenOffset()).getCharacter() / indent.length();
-                    for (int i = 0; i < level+1; i++) {
-                        insertText += indent;
-                    }
-                    insertText += FEATURE_FORMAT;
+                    insertText = (insertNode == null) ? FEATURE_FORMAT.replace("\n", "\n\t") : FEATURE_FORMAT;
+                    insertNode = (insertNode == null) ? node : insertNode;
                     break;
                 }
             }
 
-            // String insertStrRequired = null;
-            // String insertStrAll = null;
+            int insertEndOffset = (insertNode == null) ? ((DOMElement)document.getDocumentElement()).getStartTagCloseOffset()+1 : insertNode.getEnd();
```

Comment:
Maybe add a comment here explaining the end of `<server>`/`<featureManager> tag vs end of previously found `</feature>` tag?